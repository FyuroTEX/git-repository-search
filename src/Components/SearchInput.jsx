import React, { useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import debounce from 'lodash/debounce';
import { getSearchResult } from '../redux/actions/search';

import { Input } from 'semantic-ui-react';

const SearchInput = () => {
  const inputEl = useRef(null);
  const dispatch = useDispatch();
  const isLoading = useSelector((state) => state.utils.loading);
  const isError = useSelector((state) => state.utils.error);

  function request() {
    const searchQuery = inputEl.current.inputRef.current.value.trim();
    if (searchQuery.length > 0) dispatch(getSearchResult(searchQuery));
  }

  return (
    <Input
      ref={inputEl}
      loading={isLoading}
      disabled={isError}
      onChange={debounce(request, 500)}
      placeholder='Search...'
    />
  );
};

export default SearchInput;
