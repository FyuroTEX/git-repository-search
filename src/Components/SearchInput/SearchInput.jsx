import React, { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import debounce from 'lodash/debounce';
import { newRequest } from '../../redux/actions/search';

import { Input } from 'semantic-ui-react';

const SearchInput = () => {
  const dispatch = useDispatch();
  const isLoading = useSelector((state) => state.utils.loading);
  const isError = useSelector((state) => state.utils.error);

  //eslint-disable-next-line
  const delayQuery = useCallback(
    debounce(
      (val) => dispatch(newRequest({ searchRequest: val, page: 1 })),
      500
    ),
    []
  );

  const hadleChange = ({ target: { value } }) => {
    if (!value.trim().length) return;

    delayQuery(value);
  };

  return (
    <Input
      loading={isLoading}
      disabled={isError}
      onChange={hadleChange}
      placeholder='Search...'
    />
  );
};

export default SearchInput;
