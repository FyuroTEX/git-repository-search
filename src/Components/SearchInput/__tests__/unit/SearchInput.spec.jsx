import React from 'react';
import { useDispatch } from 'react-redux';
import { shallow } from 'enzyme';
import SearchInput from '../../SearchInput';

import { Input } from 'semantic-ui-react';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe('SearchInput', () => {
  const dispatch = jest.fn();
  const component = shallow(<SearchInput />);

  beforeEach(() => {
    jest.clearAllMocks();
    useDispatch.mockReturnValue(dispatch);
  });

  it('shallow SearchInput', () => {
    expect(component).toMatchSnapshot();
  });
  it('input text SearchInput', () => {
    expect(component.find(Input).exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
