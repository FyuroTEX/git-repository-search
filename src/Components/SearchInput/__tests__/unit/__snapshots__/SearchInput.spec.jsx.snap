// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput input text SearchInput 1`] = `ShallowWrapper {}`;

exports[`SearchInput shallow SearchInput 1`] = `ShallowWrapper {}`;
